pyccel/codegen/printing/ccode.py

```diff
@@ -1029,9 +1031,13 @@ def get_var_arg(arg, var):
                 return code
 
             var_list = [a.var for a in args]
-            arg_code_list = [self.function_signature(var, False) if isinstance(var, FunctionAddress) else get_var_arg(arg, var) for arg, var in zip(args, var_list)]
+            arg_code_list = [self.function_signature(var, False) if isinstance(var, FunctionAddress)
+                                else get_var_arg(arg, var) for arg, var in zip(args, var_list)]
             arg_code = ', '.join(arg_code_list)
 
+        if self._additional_args :
+            self._additional_args.pop()
```

Comment:
`_additional_args` stores all objects which are treated as arguments in C but are not arguments in python. In other words `out` arguments (maybe if it could be renamed to something clearer like `_return_args` if there are no other objects that would require this treatment). We need to be aware of these arguments as they are stored in c pointers. If  `_additional_args` is not set then the following function:
```python
def f():
    return 1, np.ones(3)
```
would be printed as:
```c
int f(int64_t Out_0001, t_ndarray Out_0002)
{
    Out_0001 = 1;
    Out_0002 = array_create(1, (int64_t[]){3}, nd_double, false);
    array_fill((double)1.0, Out_0002);
    return 0;
}
```
instead of
```c
int f(int64_t *Out_0001, t_ndarray *Out_0002)
{
    (*Out_0001) = 1;
    (*Out_0002) = array_create(1, (int64_t[]){3}, nd_double, false);
    array_fill((double)1.0, (*Out_0002));
    return 0;
}
```

Reply:
Thanks for the recap!

I am a bit confused by the fact that we call `pop()` when `_additional_args` is not an empty list. Does that mean that we expect it to contain one element only? (Which I suspect is another list)

Reply:
Indeed `_additional_args` is a list of lists. This allows it to handle functions in functions correctly